How-to guide on retriggering builds (#237)
redhat-appstudio/docs.appstudio.io: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_retriggering_build_pipelines.adoc

= Retriggering Build Pipelines

> Sometimes your build pipeline may flake out and you need to retrigger it. You can accomplish this on a pull request by adding comments to the pull request. You can accomplish this for post-merge builds by clicking a button in the UI, or by adding an annotation to your Component resource via the API.

Reviewer: ```suggestion
Occasionally, your build pipeline may fail unexpectedly, necessitating a retrigger. For pull requests, this can be achieved by adding comments. For post-merge builds, retriggering can be done either through UI, or by using the API to add an annotation to your Component resource

> See also xref:wat[Retriggering Integration Tests].

Author: To do: I need to dig up the right link.

> == Retriggering a pre-merge build on a pull request

Reviewer: ```suggestion
== Retriggering a Pre-merge build on a pull request
```

.**Prerequisite**

> - You have a build pipelinerun on a pull request that has failed and you want to re-run the build without pushing a new commit.

Reviewer: ```suggestion
- You have a build pipeline run on a pull request that has failed, and you want to re-run the build without pushing a new commit.
```

.**Procedure**

> . Add a comment to the pull request with the text `/retest` which will trigger a new build.

Reviewer: ```suggestion
. Add a comment to the pull request with the text `/retest` to trigger a new build.
```

You should see the pipelinerun start executing in the UI and in the pull request.

> NOTE: See the link:https://pipelinesascode.com/docs/guide/running/#gitops-command-on-pull-or-merge-request[pipelinesascode documentation] for more options.

Reviewer: ```suggestion
NOTE: For additional options, refer to the link:https://pipelinesascode.com/docs/guide/running/#gitops-command-on-pull-or-merge-request[pipelinesascode documentation].
```

> == Retriggering a post-merge build from your main branch from the UI

Reviewer: ```suggestion
== Retriggering a Post-merge build from your main branch from the UI
```

.**Prerequisite**

> - You have already merged a pull request, but the build failed and you want to retrigger it.

Reviewer: ```suggestion
- You have already merged a pull request, but the subsequent build failed, prompting a need for retriggering.
```

.**Procedure**

In the console, complete the following steps to retrigger the build pipeline:

. Navigate to the *Activity* > *Pipeline runs* tab.

> . Identify the pipelinerun that you want to retrigger.

Reviewer: ```suggestion
. Identify the pipeline run that you want to retrigger.
```

. Select the three dots on the right side of the table.
. Select the *Rerun* action.

> You should see the pipelinerun start executing back in the *Activity* > *Pipeline runs* tab.

Reviewer: ```suggestion
The pipeline run should resume in the *Activity* > *Pipeline runs* tab.
```

> == Retriggering a post-merge build from your main branch from the API

Reviewer: ```suggestion
== Retriggering a Post-merge build from your main branch from the API
```

.**Prerequisite**

> - You have already merged a pull request, but the build failed and you want to retrigger it.

Reviewer: ```suggestion
- You have already merged a pull request, but the subsequent build failed, prompting a need for retriggering.
```

- You have CLI access to {ProductName}. For information on obtaining CLI access, refer to  xref:../../getting-started/getting_started_in_cli.adoc[Getting started in CLI]

.**Procedure**

. Identify the *Component* whose pipeline needs to be rerun.

> . Annotate the *Component* with `build.appstudio.openshift.io/request: trigger-pac-build`.

Reviewer: ```suggestion
. Use the following command to annotate the *Component*, triggering a new build:
```
